plugins/catalog-react/src/apis/EntityPresentationApi/EntityPresentationApi.ts

```diff
@@ -0,0 +1,147 @@
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import {
+  ApiRef,
+  IconComponent,
+  createApiRef,
+} from '@backstage/core-plugin-api';
+import { Observable } from '@backstage/types';
+
+/**
+ * An API that handles how to represent entities in the interface.
+ *
+ * @public
+ */
+export const entityPresentationApiRef: ApiRef<EntityPresentationApi> =
+  createApiRef({
+    id: 'plugin.catalog.entity-presentation',
+  });
+
+/**
+ * The visual presentation of an entity reference at some point in time.
+ *
+ * @public
+ */
+export interface EntityRefPresentationSnapshot {
+  /**
+   * The ref to the entity that this snapshot represents.
+   *
+   * @remarks
+   *
+   * Note that when the input data was broken or had missing vital pieces of
+   * information, this string may contain placeholders such as "unknown". You
+   * can therefore not necessarily assume that the ref is completely valid and
+   * usable for example for forming a clickable link to the entity.
+   */
+  entityRef: string;
+  /**
+   * The entity that this snapshot represents.
+   *
+   * @remarks
+   *
+   * Note that this may be a partial entity, i.e. it may not contain all of the
+   * fields that the kind at hand normally contains, since the input to the
+   * representation renderer might have been a partial entity. So be careful
+   * when accessing fields on this.
+   */
+  entity?: Entity | undefined;
+  /**
+   * A string that can be used as a plain representation of the entity, for
+   * example in a header or a link.
+   *
+   * @remarks
+   *
+   * The title may be short and not contain all of the information that the
+   * entity holds. When rendering the primary title, you may also want to
+   * make sure to add more contextual information nearby such as the icon or
+   * secondary title, since the primary could for example just be the
+   * `metadata.name` of tne entity which might be ambiguous to the reader.
+   */
+  primaryTitle: string;
+  /**
+   * Optionally, some additional textual information about the entity, to be
+   * used as a clarification on top of the primary title.
+   *
+   * @remarks
+   *
+   * This text can for example be rendered in a tooltip or be used as a
+   * subtitle. It may not be sufficient to display on its own; it should
+   * typically be used in conjunction with the primary title. It can contain
+   * such information as the entity ref and/or a `spec.type` etc.
+   */
+  secondaryTitle?: string;
+  /**
+   * Optionally, an icon that represents the kind/type of entity.
+   *
+   * @remarks
+   *
+   * This icon should ideally be easily recognizable as the kind of entity, and
+   * be used consistently throughout the Backstage interface. It can be rendered
+   * both in larger formats such as in a header, or in smaller formats such as
+   * inline with regular text, so bear in mind that the legibility should be
+   * high in both cases.
+   */
+  Icon?: IconComponent | undefined;
```

Comment:
Some clever solution for icons 😅. The tricky bits are that we do not have an app icons API to rely on. We do not want to fall back on icons from `useApp` because that would override someone explicitly configuring the presentation API to omit icons. We do feel we need icons to be present because this change might otherwise remove too much information, as we're no longer rendering the kind in many places. We do want the icons to be configurable and a lookup table by kind would be alright to start with, but it could be nice to have more of an icon lookup function too, although that'd add the additional complexity when we don't have access to the full entity.

Reply:
fixed in [8ca3f42](https://github.com/backstage/backstage/pull/17474/commits/1f06689380810795f18b6d74829a36fd98ca3f42)

Reply:
Summary:

- `EntityRefPresentationSnapshot.Icon` can now also be `false`, signalling "i really prefer to not have any icon". `unknown` still means "i don't know or don't care, do what you want"
- `DefaultEntityPresentationApi` now accepts a `kindIcons?: Record<string, IconComponent>` which it will try to use if and only if the renderer returned undefined for the icon.